ksa_compliance/standard_doctypes/sales_invoice.py

```diff
@@ -101,3 +101,22 @@ def should_enable_zatca_for_invoice(invoice_id: str) -> bool:
 
     posting_date = frappe.db.get_value('Sales Invoice', invoice_id, 'posting_date')
     return posting_date >= start_date
+
+
+def calculate_tax_amount(self, method):
+    if self.items:
+        for item in self.items:
+            item_tax_template = frappe.get_doc("Item Tax Template", item.item_tax_template)
+            if len(item_tax_template.taxes) > 1:
+                frappe.throw(
+                    msg="""
+                        One or more items have a tax template with multiple tax accounts which is not currently 
+                        supported. Please contact the vendor.
+                        """,
+                    title="Multiple tax accounts found"
+                )
+            if item_tax_template.disabled:
+                frappe.throw("One or more items has disabled tax template", title="Disabled tax template")
+            item_tax_rate = frappe.get_value("Item Tax Template Detail", {"parent": item.item_tax_template}, ['tax_rate'])
```

Comment:
Loading a doc with `frappe.get_doc` usually loads all relevant information (e.g. child tables), so instead of hitting the database again with `get_value` we can just use `item_tax_template.taxes[0].tax_rate`